virtool/samples/models.py

```diff
@@ -0,0 +1,52 @@
+from sqlalchemy import Column, DateTime, Enum, Integer, String
+
+from virtool.pg.utils import Base, SQLEnum
+
+
+class ArtifactType(str, SQLEnum):
+    """
+    Enumerated type for possible artifact types
+
+    """
+
+    sam = "sam"
+    bam = "bam"
+    fasta = "fasta"
+    fastq = "fastq"
+    csv = "csv"
+    tsv = "tsv"
+    json = "json"
+
+
+class SampleArtifact(Base):
+    """
+    SQL model to store sample artifacts
+
+    """
+    __tablename__ = "sample_artifacts"
+
+    id = Column(Integer, primary_key=True)
+    sample = Column(String, nullable=False)
+    name = Column(String, nullable=False)
+    name_on_disk = Column(String)
+    size = Column(Integer)
+    type = Column(Enum(ArtifactType), nullable=False)
+    uploaded_at = Column(DateTime)
+
+
+class SampleReadsFile(Base):
+    """
+    SQL model to store new sample reads files
+
+    """
+    __tablename__ = "sample_reads_files"
+
+    id = Column(Integer, primary_key=True)
+    sample = Column(String, nullable=False)
```

Comment:
Should have `name` column. Will always be `reads_1.fq.gz` or `reads_2.fq.gz` at this point.

Reply:
Should I simply use `name` over `name_on_disk` since they would both be the same?

Reply:
No. The legacy type sample files are actually called `reads_1.fastq` etc.

Reply:
Oh I see. Right now it doesn't make use of a query, but should I set it up so `name` can be passed in (validated to be either `reads_1.fq.gz` or `reads_2.fq.gz`) and `name_on_disk` can be is the actual filename?